homeassistant/components/eq3btsmart/__init__.py

```diff
@@ -0,0 +1,146 @@
+    )
+
+    if device is None:
+        raise ConfigEntryNotReady(f"[{eq3_config.name}] Device could not be found")
+
+    thermostat = Thermostat(
+        thermostat_config=ThermostatConfig(
+            mac_address=mac_address,
+            name=name,
+        ),
+        ble_device=device,
+    )
+
+    eq3_config_entry = Eq3ConfigEntryData(eq3_config=eq3_config, thermostat=thermostat)
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = eq3_config_entry
+
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    entry.async_create_background_task(
+        hass, _run_thermostat(hass, entry), entry.entry_id
+    )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Handle config entry unload."""
+
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        eq3_config_entry: Eq3ConfigEntryData = hass.data[DOMAIN].pop(entry.entry_id)
+        await eq3_config_entry.thermostat.async_disconnect()
+
+    return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle config entry update."""
+
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def _run_thermostat(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Run the thermostat."""
+
+    eq3_config_entry: Eq3ConfigEntryData = hass.data[DOMAIN][entry.entry_id]
+    thermostat: Thermostat = eq3_config_entry.thermostat
+    name: str = eq3_config_entry.eq3_config.name
+    mac_address: str = eq3_config_entry.eq3_config.mac_address
+    scan_interval: int = eq3_config_entry.eq3_config.scan_interval
+
+    await _reconnect_thermostat(hass, entry)
+
+    while True:
+        try:
+            await thermostat.async_get_status()
+        except Eq3Exception as e:
+            if not thermostat.is_connected:
+                _LOGGER.error(
+                    "[%s] eQ-3 device disconnected",
+                    name,
+                )
+                async_dispatcher_send(
+                    hass,
+                    SIGNAL_THERMOSTAT_DISCONNECTED,
+                    mac_address,
+                )
+                await _reconnect_thermostat(hass, entry)
+                continue
+
+            _LOGGER.error(
+                "[%s] Error updating eQ-3 device: %s",
+                name,
+                e,
+            )
+
+        await asyncio.sleep(scan_interval)
+
+
+async def _reconnect_thermostat(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Reconnect the thermostat."""
+
+    eq3_config_entry: Eq3ConfigEntryData = hass.data[DOMAIN][entry.entry_id]
+    thermostat: Thermostat = eq3_config_entry.thermostat
+    mac_address: str = eq3_config_entry.eq3_config.mac_address
+    scan_interval: int = eq3_config_entry.eq3_config.scan_interval
+    name: str = eq3_config_entry.eq3_config.name
```

Comment:
```suggestion
    thermostat = eq3_config_entry.thermostat
    name = eq3_config_entry.eq3_config.name
    mac_address = eq3_config_entry.eq3_config.mac_address
    scan_interval = eq3_config_entry.eq3_config.scan_interval
```

The type hints are already defined in the model